packages/nimble-components/src/table/components/cell/index.ts

```diff
@@ -14,10 +20,42 @@ declare global {
  * A styled cell that is used within the nimble-table-row.
  * @internal
  */
-export class TableCell extends FoundationElement {
-    // TODO: This should be replaced with an instance of TableCellState<TCellData>
+export class TableCell<
+    TCellData extends TableRecord = TableRecord
+> extends FoundationElement {
     @observable
-    public data: TableFieldValue;
+    public data?: TableCellState<TCellData>;
+
+    @observable
+    public cellTemplate?: ViewTemplate;
+
+    @observable
+    public cellStyles?: ElementStyles;
+
+    /**
+     * @internal
+     */
+    public readonly cellContainer: HTMLElement | undefined = undefined;
+
+    private customCellView: HTMLView | undefined = undefined;
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.customCellView = this.cellTemplate?.render(
+            this.data,
+            this.cellContainer!
+        );
+    }
+
+    private dataChanged(): void {
+        this.customCellView?.bind(this.data, defaultExecutionContext);
+    }
+
+    private cellStylesChanged(): void {
+        if (this.cellStyles) {
+            this.$fastController.addStyles(this.cellStyles);
```

Comment:
Do we need to remove old styles? I don't know much about how this works, just noting the lack of symmetry.

Reply:
While we should never hit the case where we remove previous styles (as they are defined statically), I've updated the implementation to remove old styles if present.